src/core/SliceRepresentation.js

```diff
@@ -194,16 +213,18 @@ export default class SliceRepresentation extends Component {
   dataChanged() {
     if (this.props.colorDataRange === 'auto') {
       this.mapper.update();
-      const input = this.mapper.getInputData();
-      const array = input && input.getPointData()?.getScalars();
-      const dataRange = array && array.getRange();
-      if (dataRange) {
-        this.lookupTable.setMappingRange(...dataRange);
-        this.lookupTable.updateRange();
-        this.piecewiseFunction.setNodes([
-          { x: dataRange[0], y: 0, midpoint: 0.5, sharpness: 0 },
-          { x: dataRange[1], y: 1, midpoint: 0.5, sharpness: 0 },
-        ]);
+      if (this.mapper && this.mapper.getInputData()) {
```

Comment:
When would `this.mapper` be falsy? I presume we'd always want `this.mapper` to exist during the lifetime of this component.

Reply:
Hmmm, I must have added this after getting some warnings. Let me remove and see if its really necessary or not.

Reply:
Removed `this.mapper` from the if condition.